### DISCUSSION TITLE
Why is raylib Release build for Android larger than Debug build?

### DISCUSSION THREAD

**Discussion by OussamaTeyib** [raysan5/raylib#4458] [Q&A] [answered]
I built Raylib for Android using CMake and noticed a large size difference between debug and release builds. The Debug build is around 7 MB, while the Release build is 12 MB.

Here’s the command I used:

```
rm -rf Build
cmake -B Build -DCMAKE_TOOLCHAIN_FILE=%ANDROID_NDK_HOME%\build\cmake\android.toolchain.cmake -DPLATFORM=Android -DANDROID_ABI=arm64-v8a -DANDROID_PLATFORM=android-21 -DBUILD_EXAMPLES=OFF -DCMAKE_BUILD_TYPE=Release/Debug -G Ninja
cmake --build Build
```

Is it normal for the Release build to be nearly twice the size of the Debug build? Any tips on reducing the release build size?

**Comment by jestarray** [comment]
@OussamaTeyib Wait, is that a working cmake command for cross compiling to android?

**Reply by OussamaTeyib** [reply]
Yes, it’s works fine. I’m using the NDK’s built-in toolchain file to cross-compile for Android. I used the output library in an Android project, and everything worked fine.

What's more interesting is that when using the provided Makefile:
```
make PLATFORM=PLATFORM_ANDROID ANDROID_NDK=%ANDROID_NDK_HOME% ANDROID_API_VERSION=35 ANDROID_ARCH=arm64
```
I get a library size of only 2.5 MB!

So, there's a big difference between both methods.

**Reply by jestarray** [reply]
Do you know how to make build the examples and make them runnable on android phone? Cross compiling works, I'm stuck on getting them into apk form

**Reply by OussamaTeyib** [reply]
I never built raylib exemples for Android. But if you want a way to turn them into an Android app, u can use the cmake file in [this repository](https://github.com/OussamaTeyib/HelloWorld) with minimal changes.
It is a simple Android app built entirely in C. So, just replace the code in main.c with the code of the exemple, and use cmake command to compile the native code and to create the apk package.

**Reply by jestarray** [reply]
Cool, I cloned that repo and got an app working on my phone. Do you know how to load textures and such? Trying to put it in the `res` folder and then specifying `LoadTexture("res/bunny.png")` but its not working

**Reply by OussamaTeyib** [reply]
In Android, `LoadTexture("res/bunny.png")` won’t work because the `res` folder is not accessible at runtime. Instead:

1. Place the asset in the `assets` folder (not `res`).

2. Use `LoadTexture("bunny.png")` to load the asset from the `assets` folder.

<del>
2. Use `AAssetManager` from the Android NDK to open assets at runtime.

3. Load the image with `LoadImageFromMemory` and create a texture with `LoadTextureFromImage`.

Use raylib's `GetAndroidApp` to retrieve the Android App instance needed to access the asset manager.
</del>

**Reply by jestarray** [reply]
Ahh I see, I'm new to android so I didn't now it couldn't just read from directory. Where should the assets folder located? in `src/assets`? or `res/assets`? Also is that it? No need to specify the assets folder in any of the xml files?

**Reply by OussamaTeyib** [reply]
Yes, you don't need any XML configuration. The `assets` folder is typically placed alongside the `res` folder.
Remember to modify the `create_apk` target in the CMake file to include the assets in the final APK package by using the `-A` flag.

**Comment by OussamaTeyib** [comment, marked as answer]
For Android, you should use `MinSizeRel` instead of `Release`. However, the NDK's CMake toolchain always includes debug symbols in all build variants, which is why your release build ends up being large. You should ensure that your packaging tools strip these symbols from the libraries before including them in the APK (AGP do this by default).
